[CI] LegacyGeoShapeWithDocValuesQueryTests testIndexPointsFromLine failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3914 / ubuntu-2004-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/kfnynztnhzw7u)
- [elasticsearch-pull-request #29880 / part-2-fips](https://gradle-enterprise.elastic.co/s/wk3b46mczqnps)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:spatial:test" --tests "org.elasticsearch.xpack.spatial.index.query.LegacyGeoShapeWithDocValuesQueryTests.testIndexPointsFromLine" -Dtests.seed=2CC260A41BF245F -Dtests.locale=ar-OM -Dtests.timezone=America/Vancouver -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testIndexPointsFromLine'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.spatial.index.query.LegacyGeoShapeWithDocValuesQueryTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Count is 2 hits but 3 was expected.  Total shards: 1 Successful shards: 1 & 0 shard failures:
```

**Issue Reasons:**
- [main] 2 failures in test testIndexPointsFromLine (9.1% fail rate in 22 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -176,9 +176,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/112421
 - class: org.elasticsearch.indices.mapping.UpdateMappingIntegrationIT
   issue: https://github.com/elastic/elasticsearch/issues/112423
-- class: org.elasticsearch.xpack.spatial.index.query.LegacyGeoShapeWithDocValuesQueryTests
-  method: testIndexPointsFromLine
-  issue: https://github.com/elastic/elasticsearch/issues/112438
 - class: org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialCentroidTests
   method: "testAggregateIntermediate {TestCase=<geo_point> #2}"
   issue: https://github.com/elastic/elasticsearch/issues/112461

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -185,9 +185,6 @@ tests:
 - class: org.elasticsearch.xpack.inference.external.http.RequestBasedTaskRunnerTests
   method: testLoopOneAtATime
   issue: https://github.com/elastic/elasticsearch/issues/112471
-- class: org.elasticsearch.xpack.spatial.index.query.LegacyGeoShapeWithDocValuesQueryTests
-  method: testIndexPointsFromPolygon
-  issue: https://github.com/elastic/elasticsearch/issues/112464
 
 # Examples:
 #

--- a/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LegacyGeoShapeWithDocValuesQueryTests.java
+++ b/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LegacyGeoShapeWithDocValuesQueryTests.java
@@ -27,6 +27,7 @@ import org.elasticsearch.xcontent.XContentFactory;
 import org.elasticsearch.xpack.spatial.LocalStateSpatialPlugin;
 
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 
@@ -219,5 +220,18 @@ public class LegacyGeoShapeWithDocValuesQueryTests extends GeoShapeQueryTestCase
             .get();
 
         assertHitCount(client().prepareSearch(defaultIndexName).setQuery(geoShapeQuery("alias", multiPoint)), 1L);
+    }
+
+    protected boolean ignoreLons(double[] lons) {
+        return Arrays.stream(lons).anyMatch(v -> v == 180);
+    }
+
+    public void testIndexPointsFromLine() throws Exception {
+        super.testIndexPointsFromLine();
+    }
+
+    @Override
+    public void testIndexPointsFromPolygon() throws Exception {
+        super.testIndexPointsFromPolygon();
     }
 }

--- a/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LegacyGeoShapeWithDocValuesQueryTests.java
+++ b/x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LegacyGeoShapeWithDocValuesQueryTests.java
@@ -225,13 +225,4 @@ public class LegacyGeoShapeWithDocValuesQueryTests extends GeoShapeQueryTestCase
     protected boolean ignoreLons(double[] lons) {
         return Arrays.stream(lons).anyMatch(v -> v == 180);
     }
-
-    public void testIndexPointsFromLine() throws Exception {
-        super.testIndexPointsFromLine();
-    }
-
-    @Override
-    public void testIndexPointsFromPolygon() throws Exception {
-        super.testIndexPointsFromPolygon();
-    }
 }

